pkg/kubectl/cmd/annotate.go

```diff
@@ -189,31 +189,23 @@ func (o AnnotateOptions) RunAnnotate(f cmdutil.Factory, cmd *cobra.Command) erro
 	changeCause := f.Command(cmd, false)
 
 	includeUninitialized := cmdutil.ShouldIncludeUninitialized(cmd, false)
-	b := f.NewBuilder().
-		ContinueOnError().
-		NamespaceParam(namespace).DefaultNamespace().
-		FilenameParam(enforceNamespace, &o.FilenameOptions).
-		IncludeUninitialized(includeUninitialized).
-		Flatten()
 
+	var b *resource.Builder
 	if !o.local {
-		// call this method here, as it requires an api call
-		// and will cause the command to fail when there is
-		// no connection to a server
-		mapper, typer, err := f.UnstructuredObject()
-		if err != nil {
-			return err
-		}
-
-		b = b.LabelSelectorParam(o.selector).
-			Unstructured(f.UnstructuredClientForMapping, mapper, typer).
+		b = f.NewUnstructuredBuilder().
```

Comment:
This code block `if !o.local {...} else {...}` happens four times.  Any way to have just one copy (even if it means casting)? 

Also, FWIW, for full if/else statements its usually easier to read with the true case first.